docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,921 @@
+    
+    // MinReadySeconds is the minimum number of seconds for which a Node for a newly created machine should be ready before considering the replica available.
+    // Defaults to 0 (machine will be considered available as soon as the Node is ready)
+    // +optional
+    MinReadySeconds int32 `json:"minReadySeconds,omitempty"`
+
+    // If specified, all readiness gates will be evaluated for Machine readiness.
+    // A Machine is ready when `InfrastructureReady`, `NodeHealthy` and `HealthCheckSucceeded` (if present) are "True"; 
+    // if other conditions are defined this field, those conditions should be "True" as well for the Machine to be ready.
+    // +optional
+    // +listType=map
+    // +listMapKey=conditionType
+    ReadinessGates []MachineReadinessGate `json:"readinessGates,omitempty"`
+
+    // Other fields...
+}
+
+// MachineReadinessGate contains the reference to a Machine condition to be used as readiness gates.
+type MachineReadinessGate struct {
+    // ConditionType refers to a condition in the Machine's condition list with matching type.
+    // Note: Both  Cluster API conditions or conditions added by 3rd party controller can be used as readiness gates.
+    ConditionType string `json:"conditionType"`
+}
+```
+
+| v1beta1 (current)      | v1Beta2 (tentative Q1 2025) | v1beta1 removal (tentative Q1 2026) |
+|------------------------|-----------------------------|-------------------------------------|
+| `ReadinessGates` (new) | `ReadinessGates`            | `ReadinessGates`                    |
+| other fields...        | other fields...             | other fields...                     |
+
+Notes:
+- Both `MinReadySeconds` and `ReadinessGates` should be treated as other in-place propagated fields (changing this should not trigger rollouts).
+- Similarly to Pod's `ReadinessGates`, also Machine's `ReadinessGates` accept only conditions with positive polarity; 
+  The Cluster API project might revisit this in future to stay aligned with Kubernetes or if there are use cases justifying this change.
+
+#### Machine Print columns
+
+| Current           | To be                         |
+|-------------------|-------------------------------|
+| `NAME`            | `NAME`                        |
+| `CLUSTER`         | `CLUSTER`                     |
+| `NODE NAME`       | `PAUSED` (new) (*)            |
+| `PROVIDER ID`     | `NODE NAME`                   |
+| `PHASE` (deleted) | `PROVIDER ID`                 |
+| `AGE`             | `READY` (new)                 |
+| `VERSION`         | `AVAILABLE` (new)             |
+|                   | `UP TO DATE` (new)            |
+|                   | `AGE`                         |
+|                   | `OS-IMAGE` (new) (*)          |
+|                   | `KERNEL-VERSION` (new) (*)    |
+|                   | `CONTAINER-RUNTIME` (new) (*) |
+
+TODO: figure out if can `INTERNAL-IP` (new) (*),   `EXTERNAL-IP` after `VERSION` / before `OS-IMAGE`?  (similar to Nodes...).
```

Comment:
+++1 and would love if we can get all of these in -o wide

Reply:
Thanks for the feedback! 💯  agree!

Resolving to keep the number of open threads under control 😅